plan/decorrelate.go

```diff
@@ -102,8 +133,29 @@ func (s *decorrelateSolver) optimize(p LogicalPlan, _ context.Context, _ *idAllo
 				return proj, nil
 			}
 			return s.optimize(p, nil, nil)
+		} else if agg, ok := innerPlan.(*Aggregation); ok {
+			if apply.canPullUpAgg() && agg.canPullUp() {
+				innerPlan = agg.children[0].(LogicalPlan)
```

Comment:
line 90's innerPlan is apply.children[1], at this place , it will be replaced by agg's child.